File: leetcode2025/2025_02/leetcode2025-02-24.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class OrderedStream {
private:
    vector<string> v;
    int ptr;
public:
    OrderedStream(int n) {
        v.resize(n+1, "");
        ptr = 1;
    }
    
    vector<string> insert(int idKey, string value) {
        v[idKey] = value;
        int tmpid = ptr;
        vector<string> res;
        if (v[tmpid]!=""){
            res.push_back(v[tmpid]);
        }
        while (tmpid<v.size()-1 && v[tmpid]!="" && v[tmpid+1]!=""){
            tmpid++;
            res.push_back(v[tmpid]);
        }
        ptr += res.size();
        return res;
    }
};

int main(){
    OrderedStream os(5);
    vector<int> indxs = {3,1,2,5,4};
    vector<string> strs = {"ccccc","aaaaa","bbbbb","eeeee","ddddd"};
    for (int i=0; i<indxs.size(); i++){
        vector<string> res = os.insert(indxs[i], strs[i]);
        for (string& j : res){
            cout << j << " ";
        }
        cout << endl;
    }
}

File: leetcode2025/2025_02/leetcode2025-02-24.py
from typing import List

class OrderedStream:

    def __init__(self, n: int):
        self.lst = [""]*(n+1)
        self.ptr = 1

    def insert(self, idKey: int, value: str) -> List[str]:
        self.lst[idKey] = value
        tmpid = self.ptr
        res = []
        if self.lst[tmpid]!="":
            res.append(self.lst[tmpid])
        while tmpid<len(self.lst)-1 and self.lst[tmpid]!="" and self.lst[tmpid+1]!="":
            tmpid += 1
            res.append(self.lst[tmpid])
        self.ptr += len(res)
        return res

if __name__ == "__main__":
    os = OrderedStream(5)
    pairs = [[3,"ccccc"],[1,"aaaaa"],[2,"bbbbb"],[5,"eeeee"],[4,"ddddd"]]
    for p in pairs:
        print(os.insert(p[0], p[1]))
